pkg/fleets/controller_test.go

```diff
@@ -752,20 +752,20 @@ func TestControllerUpdateFleetCounterStatus(t *testing.T) {
 			ua := action.(k8stesting.UpdateAction)
 			fleet := ua.GetObject().(*agonesv1.Fleet)
 
-			assert.Equal(t, int64(1100), fleet.Status.Counters["fullCounter"].AllocatedCount)
-			assert.Equal(t, int64(1100), fleet.Status.Counters["fullCounter"].AllocatedCapacity)
-			assert.Equal(t, int64(1100), fleet.Status.Counters["fullCounter"].Capacity)
-			assert.Equal(t, int64(1100), fleet.Status.Counters["fullCounter"].Count)
-
-			assert.Equal(t, int64(11), fleet.Status.Counters["anotherCounter"].AllocatedCount)
-			assert.Equal(t, int64(20), fleet.Status.Counters["anotherCounter"].AllocatedCapacity)
-			assert.Equal(t, int64(200), fleet.Status.Counters["anotherCounter"].Capacity)
-			assert.Equal(t, int64(42), fleet.Status.Counters["anotherCounter"].Count)
-
-			assert.Equal(t, int64(21), fleet.Status.Counters["thirdCounter"].AllocatedCount)
-			assert.Equal(t, int64(30), fleet.Status.Counters["thirdCounter"].AllocatedCapacity)
-			assert.Equal(t, int64(400), fleet.Status.Counters["thirdCounter"].Capacity)
-			assert.Equal(t, int64(21), fleet.Status.Counters["thirdCounter"].Count)
+			assert.Equal(t, agonesv1.SafeAdd(1000, 100), fleet.Status.Counters["fullCounter"].AllocatedCount)
```

Comment:
Here we want to leave `assert.Equal(t, int64(21), fleet.Status.Counters["thirdCounter"].Count)`. The `err := c.updateFleetStatus(ctx, fleet)` will call the SafeAdd as part of the method, so that's part of what we're testing.